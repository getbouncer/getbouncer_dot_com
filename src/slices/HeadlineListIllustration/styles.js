import styled from '@emotion/styled'
import { breakpoints, theme } from '~styles'

export const Wrapper = styled.section`
  margin: 0 0 150px;

  @media ${breakpoints.smartphone} {
    margin: 0 0 70px;
  }
`

export const Headline = styled.div`
  text-align: center;

  h2 {
    background: ${theme.textGradient};
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    display: inline-block;
    font-size: 2.85rem;
    line-height: 4rem;
  }

  @media ${breakpoints.smartphone} {
    text-align: left;
  }
`

export const Content = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 75px;
  padding: 0 30px 0 75px;

  @media ${breakpoints.smartphone} {
    display: block;
    margin-top: 42px;
    padding: 0;
  }
`

export const Grid = styled.div`
  max-width: 450px;
`

export const Item = styled.div`
  position: relative;
  margin-left: 34px;

  &:not(:last-child) {
    margin-bottom: 50px;
  }

  @media ${breakpoints.smartphone} {
    margin-left: 0;
  }
`

export const ItemTitle = styled.h3`
  margin-bottom: 24px;
`

export const Icon = styled.img`
  left: -107px;
  position: absolute;
  width: 72px;

  @media ${breakpoints.smartphone} {
    position: static;
    margin-bottom: 30px;
  }
`

export const Illustration = styled.img`
  align-self: center;
  width: 40%;
  max-width: 600px;

  @media ${breakpoints.smartphone} {
    display: block;
    margin: 40px auto 0;
    width: 100%;
  }
`
