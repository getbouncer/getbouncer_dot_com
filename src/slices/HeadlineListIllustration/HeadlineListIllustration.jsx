import React from 'react'
import PropTypes from 'prop-types'
import { Content, Grid, Item, Icon, Illustration, ItemTitle, Headline, Wrapper } from './styles'
import Container from '~globals/Container'

const HeadlineListIllustration = ({ input }) => {
  const { items, primary } = input

  return (
    <Container>
      <Wrapper>
        <Headline dangerouslySetInnerHTML={{ __html: primary.headline.html }} />
        <Content>
          <Grid>
            {items.map(item => (
              <Item key={item.item_title}>
                <Icon src={item.item_icon.url} alt={item.item_icon.alt} />
                <ItemTitle>{item.item_title}</ItemTitle>
                <div dangerouslySetInnerHTML={{ __html: item.item_description.html }} />
              </Item>
            ))}
          </Grid>
          <Illustration src={primary.illustration.url} alt={primary.illustration.alt} />
        </Content>
      </Wrapper>
    </Container>
  )
}

HeadlineListIllustration.propTypes = {
  input: PropTypes.shape().isRequired,
}

export default HeadlineListIllustration
