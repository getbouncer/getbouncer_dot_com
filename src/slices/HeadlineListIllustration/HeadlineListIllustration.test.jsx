import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, it } from 'jest'
import HeadlineListIllustration from './HeadlineListIllustration'

describe('HeadlineListIllustration', () => {
  it('renders correctly', () => {
    const data = {
      primary: {
        headline: {
          html: '<h1>headline</h1>',
        },
        illustration: {
          alt: 'illustration',
          url: 'https://www.example.com/illustration.png',
        },
      },
      items: [
        {
          item_title: 'item 1 title',
          item_icon: {
            alt: 'item 1 icon',
            url: 'https://www.example.com/item_1_icon.svg',
          },
          item_description: {
            html: '<p>item description</p>',
          },
        },
        {
          item_title: 'item 2 title',
          item_icon: {
            alt: 'item 2 icon',
            url: 'https://www.example.com/item_2_icon.svg',
          },
          item_description: {
            html: '<p>item description</p>',
          },
        },
      ],
    }
    const tree = renderer
      .create(<HeadlineListIllustration input={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
