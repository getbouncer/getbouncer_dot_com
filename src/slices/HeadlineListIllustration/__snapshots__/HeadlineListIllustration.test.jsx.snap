// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadlineListIllustration renders correctly 1`] = `
<div
  className="css-1fywgw4"
>
  <section
    className="css-vp3p26"
  >
    <div
      className="css-1mci2r3"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1>headline</h1>",
        }
      }
    />
    <div
      className="css-1ma0d39"
    >
      <div
        className="css-1bivl2p"
      >
        <div
          className="css-1r631za"
        >
          <img
            alt="item 1 icon"
            className="css-10n7if5"
            src="https://www.example.com/item_1_icon.svg"
          />
          <h3
            className="css-exhu38"
          >
            item 1 title
          </h3>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>item description</p>",
              }
            }
          />
        </div>
        <div
          className="css-1r631za"
        >
          <img
            alt="item 2 icon"
            className="css-10n7if5"
            src="https://www.example.com/item_2_icon.svg"
          />
          <h3
            className="css-exhu38"
          >
            item 2 title
          </h3>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>item description</p>",
              }
            }
          />
        </div>
      </div>
      <img
        alt="illustration"
        className="css-8uca2l"
        src="https://www.example.com/illustration.png"
      />
    </div>
  </section>
</div>
`;
